src/query/src/part_sort.rs

```diff
@@ -827,6 +933,32 @@ impl PartSortStream {
         sorted_batch.map(|x| if x.num_rows() == 0 { None } else { Some(x) })
     }
 
+    /// Helper to handle outputting a sorted batch:
+    /// - Updates total_emitted
+    /// - Updates limit
+    /// - Checks for early termination based on threshold comparison with next group
+    fn handle_sorted_output(
+        &mut self,
+        sorted_batch: DfRecordBatch,
+    ) -> datafusion_common::Result<DfRecordBatch> {
+        let batch_rows = sorted_batch.num_rows();
+        self.total_emitted += batch_rows;
+
+        // Update limit
+        self.limit = self.limit.map(|l| l.saturating_sub(batch_rows));
```

Comment:
is it correct to sub limit here? Like ASC limit = 4
Group 1 [0, 20): [0,5,14,15]
Group 2 [2, 21): [1,3,4,7]
updating limit here after done with group 1 still make group 2 can't be output?